lib/system/threads.nim

```diff
@@ -491,8 +491,6 @@ else:
     ## waits for every thread in `t` to finish.
     for i in 0..t.high: joinThread(t[i])
 
-when false:
-  # XXX a thread should really release its heap here somehow:
```

Comment:
Nope, thats mistake. To enable `destroyThread` we really need to release heap somehow.